pkg/apis/kops/cluster.go

```diff
@@ -233,6 +233,33 @@ type ClusterSpec struct {
 
 	// Networking configuration
 	Networking *NetworkingSpec `json:"networking,omitempty"`
+
+	// API field controls how the API is exposed outside the cluster
+	API *PublishSpec `json:"api,omitempty"`
+}
+
+type PublishSpec struct {
```

Comment:
`AccessSpec` would work.    It may be overkill, but this isn't necessarily just the API.

That said, probably YAGNI, so we could just go with `KubernetesApiAccessSpec`.

It is invisible externally - it isn't really part of our schema unless you're driving it from go.